extensions/kind/src/create-cluster.spec.ts

```diff
@@ -606,3 +637,76 @@ test('check that auditItems returns multiple error messages when both ports are
   expect(checks.records[1].record).toContain('Invalid HTTPS Port 888888');
   expect(checks.records[1].record).toContain('Please enter a port number between 0 and 65535.');
 });
+
+describe('waitForCoreDNSReady', () => {
+  test('should check all components using Kubernetes client', async () => {
+    // Uses default mocks from global beforeEach (all components ready)
+    await waitForCoreDNSReady();
+
+    // Verify nodes were checked
+    expect(vi.mocked(k8s.CoreV1Api.prototype.listNode)).toHaveBeenCalledTimes(1);
```

Comment:
remark: you do not need to use the `vi.mocked` inside a expect, vitest is smart enough to understand this is a mock when passing a function

```suggestion
    expect(k8s.CoreV1Api.prototype.listNode).toHaveBeenCalledTimes(1);
```